site/docs/v1/tech/samlv2/zendesk.adoc

```diff
@@ -5,19 +5,103 @@ description: Setting up Zendesk to login using FusionAuth via SAML v2
 ---
 
 == Overview
+Zendesk allows customers to sign into their Zendesk accounts using a SAML identity provider. This document covers the configuration necessary to get Zendesk working with FusionAuth as the identity provider using SAML v2.
 
-=== Enable SAML SSO in Zendesk
-1. https://<your_domain>.zendesk.com/admin/security/sso
-2. Select SAML --> Edit
-3. Check enabled
-4. SAML SSO URL = https://login.fusionauth.io/samlv2/login/f32a81a4-e5be-93f2-cb34-ec605ed7f708
-4. Certificate fingerprint = 26:81:09:18:CD:5D:D4:3B:91:29:2D:92:38:5F:D3:5E:9E:F7:AD:7C:BF:A2:E6:BF:68:F2:D3:45:85:68:85:1F
-4. Remote logout URL = https://login.fusionauth.io/samlv2/logout/50c9e259-177c-43e1-b6b9-66b04f31a743
-5. Save
+We'll also set up roles in FusionAuth to automatically grant agent dashboard access in Zendesk.
 
+=== Prerequisites
 
-=== Enable SSO for Staff or End users
+This document assumes you have a running instance of FusionAuth and a working Zendesk application. You will also need admin accounts for both to configure them correctly.
+
+Additionally, you'll need to know your Zendesk URL. It may be something like: `https://fusionauth-example.zendesk.com`. We'll use that value for this document, but please replace it with the correct URL.
+
+Finally, you'll need a FusionAuth user to sign in to Zendesk as. You can use an existing user or create one.
+
+=== Configure FusionAuth
+
+First, we want to create a new application, `Zendesk`, in FusionAuth. Navigate to [breadcrumb]#Applications# and create an application. Create two roles as well:
+
+* Create a role called `agent`.
+* Create a role called `admin`.
+
+When you are done, your screen should look like this just before saving:
+
+image::samlv2/zendesk/create-application-and-roles.png[Creating FusionAuth application with Zendesk roles.,width=1200,role=shadowed]
+
+Next, we'll create a lambda to take FusionAuth role information and populate the SAML assertion that Zendesk will receive.
+
+* Create a lambda called `Populate Zendesk roles`. 
+* The type should be `SAML v2 Populate`. 
+* Set the body to:
+[source,javascript]
+----
+function populate(samlResponse, user, registration) {
+  samlResponse.assertion.attributes['role'] = registration.roles || ['user'];
+}
+----
```

Comment:
Can we provide one sentence after this source code example of what the code does? I would assume it is setting the SAML response role assertion equal to the User's registered roles, with a default role of `user`. 

I'm not sure the default `['user']` will ever be used? The `registration.roles` will never be null, but it may be empty. I think the `||` will only use the right hand side if the value is `undefined` or `falsey`?

Reply:
Thanks, I corrected.